1) No need to use bash to resolve the symlink if flexlink is built
as a Cygwin executable.

2) If built for Cygwin itself, not only is it unnecessary to use cygpath,
but with Cygwin 1.7, ">NUL" is understood as a redirect to file "NUL"
instead of the Windows null device (equiv. /dev/null).

--- origsrc/flexdll/reloc.ml	2009-08-07 01:45:20.611397000 -0500
+++ src/flexdll/reloc.ml	2009-08-07 01:52:12.639945100 -0500
@@ -163,7 +163,7 @@ let gcclib () =
   | `MINGW -> "-mno-cygwin "
   | _ -> ""
   in
-  Filename.dirname (get_output1 ~use_bash:(!toolchain = `CYGWIN) (Printf.sprintf "gcc %s-print-libgcc-file-name" extra))
+  Filename.dirname (get_output1 ~use_bash:(!toolchain = `CYGWIN && Sys.os_type != "Cygwin") (Printf.sprintf "gcc %s-print-libgcc-file-name" extra))
 
 let file_exists fn =
   if Sys.file_exists fn && not (Sys.is_directory fn) then Some fn
@@ -1032,11 +1032,10 @@ let main () =
 
   use_cygpath :=
     begin
-      match !toolchain, !cygpath_arg with
-      | _, `Yes -> true
-      | _, `No -> false
-      | (`CYGWIN|`MINGW), `None -> (Sys.command "cygpath -v 2>NUL >NUL" = 0)
-      | (`MSVC|`LIGHTLD), `None -> false
+      match Sys.os_type, !toolchain, !cygpath_arg with
+      | _, _, `Yes -> true
+      | "Win32", (`CYGWIN|`MINGW), `None -> (Sys.command "cygpath -v 2>NUL >NUL" = 0)
+      | _, _, _ -> false
     end;
 
 
